src/platform/ESP32/ConnectivityManagerImpl_WiFi.cpp

```diff
@@ -780,7 +780,7 @@ void ConnectivityManagerImpl::ChangeWiFiStationState(WiFiStationState newState)
         ChipLogProgress(DeviceLayer, "WiFi station state change: %s -> %s", WiFiStationStateToStr(mWiFiStationState),
                         WiFiStationStateToStr(newState));
         mWiFiStationState = newState;
-        SystemLayer().ScheduleLambda([]() { NetworkCommissioning::ESPWiFiDriver::GetInstance().OnNetworkStatusChange(); });
+        NetworkCommissioning::ESPWiFiDriver::GetInstance().OnNetworkStatusChange();
```

Comment:
Is the change worth it? I am unsure about sideffects, however this may have been done on purpose, usually to decrease stack usage or remove recursive calls (can OnNetworkStatusChange eventually call ChangeWiFiStationState?). 

I am wondering if the overhead saved by not scheduling the lambda is worth it in investigating if this is a correct change or not.

Reply:
I have checked that there ate no recursive calls i.e this scenario of  `OnNetworkStatusChange eventually call ChangeWiFiStationState does not hold`. Although the ScheduleLamda might decrease the stack usage to some extent, but scheduling it also may fill up the platform queue due to lambda events  and the application crashes due to PostEventOrDie in case of failure to post event to the platfiorm queue .
@wqx6, can you please vaildate this as well?

Reply:
Ok, please add this to the summary as to "why". It says "overhead" however it does not explain what overhead. I believe saying that this is done to decrease the size of the event queue on the platform would be useful.

Reply:
Done.